### PR TITLE
Add FV support for indicators

### PR DESCRIPTION
**Pull request by cpgr**
Enable ElementalIndicators to work with FV variables.

Refs #16069

TODO: Fix InternalSideIndicators too

### CODE REVIEW

**Comment by cpgr** [comment]
Can I get a hand @rwcarlsen or @lindsayad? I can't get the indicators derived from `InternalSideIndicator` to work with FV variables. I suspect that I haven't got the correct neighbour variables, but haven't been able to get it to work despite trying a number of things. 

There test `tests/indicators/value_jump_indicator/value_jump_indicator_fv.i` currently fails with `*** ERROR ***
unordered_map::at: key not found`

**Comment by rwcarlsen** [comment]
I think your issue is that MooseVariable is actually just an alias for MooseVariableFE<Real>.  So the short answer is an FV version will not be able to share the same class/code like this as the FE version.

**Comment by rwcarlsen** [comment]
At a minimum - your indicator header file needs to have `MooseVariableField<Real>`.  To give more detailed feedback, I'd need to know exactly where that error message is coming from - might require commenting out some exception catching.

**Comment by cpgr** [comment]
I think I have the `InternalSideIndicators` like `ValueJumpIndicator` working now

File: framework/include/indicators/ElementIndicator.h [hunk 1]
```diff
@@ -52,7 +52,7 @@ class ElementIndicator : public Indicator,
   const MooseArray<Real> & _JxW;
   const MooseArray<Real> & _coord;
 
-  MooseVariable & _var;
+  // MooseVariable & _var;
```
Comment (rwcarlsen) [inline framework/include/indicators/ElementIndicator.h:55]: remove this?

File: framework/src/indicators/InternalSideIndicator.C [hunk 2]
```diff
@@ -67,7 +67,8 @@ InternalSideIndicator::InternalSideIndicator(const InputParameters & parameters)
 
     _boundary_id(parameters.get<BoundaryID>("_boundary_id")),
 
-    _var(_subproblem.getStandardVariable(_tid, parameters.get<VariableName>("variable"))),
+    _var(static_cast<MooseVariableField<Real> &>(
```
Comment (rwcarlsen) [inline framework/src/indicators/InternalSideIndicator.C:70]: The indicator inherits from MooseVariableInterface - which provides methods you should be using to get the var.  It provides a mooseVariableField function that should give you what you need without resorting to casting.

**Review by rwcarlsen — CHANGES_REQUESTED** [review]

File: framework/include/indicators/ElementIndicator.h (hunk 1 above)
Comment (rwcarlsen) [reply framework/include/indicators/ElementIndicator.h:55]: Or make it `MooseVariableField<Real>` - is probably the right solution (and don't forget to initialize it in the `.C` file.

Comment (cpgr) [reply framework/include/indicators/ElementIndicator.h]: I did the second one, and then used it for `_u` etc.

File: framework/src/indicators/InternalSideIndicator.C (hunk 2 above)
Comment (cpgr) [reply framework/src/indicators/InternalSideIndicator.C]: Thanks @rwcarlsen  - this was the bit I needed to get it working without having to use the cast!

**Comment by rwcarlsen** [comment]
Looks like it's going to pass - yay.

**Review by rwcarlsen — APPROVED** [review]

**Pull request merged**